src/gordon_gcp/plugins/janitor/gpubsub_publisher.py

```diff
@@ -247,14 +247,15 @@ def publish(self, message):
                 break
             context = base_context.copy()
             context['action'] = change_message['action']
-            await self.metrics.incr('change-msg-recv', context=context)
             try:
                 await self.publish(change_message)
-                await self.metrics.incr(
-                    'change-msg-publish', context=context)
+                context['result'] = 'published'
             except Exception as e:  # todo
                 logging.error('Exception while trying to publish message to '
                               f' pubsub: {e}')
+                context['result'] = 'error'
```

Comment:
Does this metric/context include enough information about the error/exception that occurred ?

Reply:
I think so, because we can (should?) use the metric to know that *something* happened, but use the logs to debug, rather than trying to use a metric to work for both. Since we don't have any expected error causes, the only thing we could use would be something generic like class name, but I'm not sure how useful something like "NameError" or "KeyError" would be on it own. We'd probably still end up needing to check the logs to debug.

Reply:
:+1: That makes sense to me from an operations point of view.